source/ui/REIXS/REIXSActionBasedControlEditor.cpp

```diff
@@ -42,19 +42,9 @@ void REIXSActionBasedControlEditor::onNewSetpointChosen(double value)
 	AMControlInfo setpoint = control_->toInfo();
 	setpoint.setValue(value);
 
-	if(okToRunInBackground_) {
-		AMAction3* action = new AMControlMoveAction3(new AMControlMoveActionInfo3(setpoint), control_);
-		connect(action, SIGNAL(succeeded()), action, SLOT(deleteLater()));
-		connect(action, SIGNAL(failed()), action, SLOT(deleteLater()));
-		connect(action, SIGNAL(cancelled()), action, SLOT(deleteLater()));
-		action->start();
-	}
-	else {
-
-		AMAction3* action = new AMControlMoveAction3(new AMControlMoveActionInfo3(setpoint), control_);
-		connect(action, SIGNAL(succeeded()), action, SLOT(deleteLater()));
-		connect(action, SIGNAL(failed()), action, SLOT(deleteLater()));
-		connect(action, SIGNAL(cancelled()), action, SLOT(deleteLater()));
-		action->start();
-	}
+	AMAction3* action = new AMControlMoveAction3(new AMControlMoveActionInfo3(setpoint), control_);
```

Comment:
You should use AMActionSupport for building control move actions. It is cleaner.